### PR TITLE
DSND-3116: Pass in deltaAt to API call

### PR DESCRIPTION
**Pull request by kinpang-CH**
* Extract delta_at from delta and pass in API call
* Change processor method to use company number located in delta instead
* Awaiting local testing when API changes are completed

Resolves [DSND-3116](https://companieshouse.atlassian.net/browse/DSND-3116)

[DSND-3116]: https://companieshouse.atlassian.net/browse/DSND-3116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 1]
```diff
@@ -157,8 +159,8 @@ private void handleResponse(final HttpStatus httpStatus)
     /**
      * Invoke Charges Data API to update charges database.
      */
-    private ApiResponse<Void> deleteCharge(String chargeId) {
-        return apiClientService.deleteCharge("0", chargeId);
+    private ApiResponse<Void> deleteCharge(String companyNumber, String chargeId, String deltaAt) {
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java:162]: Can this private method be removed? It seems pretty redundant

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java [hunk 2]
```diff
@@ -87,6 +87,8 @@ public void processDelta(Message<ChsDelta> chsDelta) {
     public void processDelete(Message<ChsDelta> chsDelta) {
         final ChargesDeleteDelta chargesDeleteDelta =
                 mapToChargesDelta(chsDelta.getPayload(), ChargesDeleteDelta.class);
+        String deltaAt = chargesDeleteDelta.getDeltaAt();
+        String companyNumber = chargesDeleteDelta.getCompanyNumber();
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java:91]: company_number should be added to the `DataMapHolder`

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java (hunk 1 above)
Comment (kinpang-CH) [reply src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java]: Addressed in latest commit, also removed the one for upsert

File: src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java (hunk 2 above)
Comment (kinpang-CH) [reply src/main/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessor.java:92]: Addressed in latest commit

**Review by cgregoryCH — APPROVED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Pull request merged**